Warnings when compiling tests

When compiling the tests, I get the following warnings:

```
In file included from test/option_parser.c:2:
test/option_parser.c: In function ‘test_string_to_int_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~

(...)
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_list_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:432:17: note: in expansion of macro ‘ASSERT_FALSEm’
  432 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, invalid_inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:428:14: note: ‘buf’ declared here
  428 |         char buf[256];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
At top level:
cc1: note: unrecognized command-line option ‘-Wno-gnu-zero-variadic-macro-arguments’ may have been intended to silence earlier diagnostics
```

<details>
<summary>
make test-valgrind

</summary>

```
cc -o test/dbus.o -c test/dbus.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/draw.o -c test/draw.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/dunst.o -c test/dunst.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/helpers.o -c test/helpers.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/icon-lookup.o -c test/icon-lookup.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/icon.o -c test/icon.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/ini.o -c test/ini.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/input.o -c test/input.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/log.o -c test/log.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/markup.o -c test/markup.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/menu.o -c test/menu.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/misc.o -c test/misc.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/notification.o -c test/notification.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/option_parser.o -c test/option_parser.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
In file included from test/option_parser.c:2:
test/option_parser.c: In function ‘test_string_to_int_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:172:17: note: in expansion of macro ‘ASSERT_FALSEm’
  172 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:169:14: note: ‘buf’ declared here
  169 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_double_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:231:17: note: in expansion of macro ‘ASSERT_FALSEm’
  231 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:228:14: note: ‘buf’ declared here
  228 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_enum’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:458:24: note: in expansion of macro ‘GREATEST_FAILm’
  458 |         if (!(COND)) { GREATEST_FAILm(MSG); }                           \
      |                        ^~~~~~~~~~~~~~
test/greatest.h:1179:24: note: in expansion of macro ‘GREATEST_ASSERTm’
 1179 | #define ASSERTm        GREATEST_ASSERTm
      |                        ^~~~~~~~~~~~~~~~
test/option_parser.c:320:17: note: in expansion of macro ‘ASSERTm’
  320 |                 ASSERTm(buf, set_from_string(&val, s, ellipsize_enum_data[i].string));
      |                 ^~~~~~~
test/option_parser.c:315:14: note: ‘buf’ declared here
  315 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_enum_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:349:17: note: in expansion of macro ‘ASSERT_FALSEm’
  349 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, invalid_inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:345:14: note: ‘buf’ declared here
  345 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_boolean_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:299:17: note: in expansion of macro ‘ASSERT_FALSEm’
  299 |                 ASSERT_FALSEm(buf, success);
      |                 ^~~~~~~~~~~~~
test/option_parser.c:294:14: note: ‘buf’ declared here
  294 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_time_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:501:17: note: in expansion of macro ‘ASSERT_FALSEm’
  501 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, invalid_inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:498:14: note: ‘buf’ declared here
  498 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_sepcolor_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:634:17: note: in expansion of macro ‘ASSERT_FALSEm’
  634 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:631:14: note: ‘buf’ declared here
  631 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_length_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:722:17: note: in expansion of macro ‘ASSERT_FALSEm’
  722 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:719:14: note: ‘buf’ declared here
  719 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_list’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:458:24: note: in expansion of macro ‘GREATEST_FAILm’
  458 |         if (!(COND)) { GREATEST_FAILm(MSG); }                           \
      |                        ^~~~~~~~~~~~~~
test/greatest.h:1179:24: note: in expansion of macro ‘GREATEST_ASSERTm’
 1179 | #define ASSERTm        GREATEST_ASSERTm
      |                        ^~~~~~~~~~~~~~~~
test/option_parser.c:389:17: note: in expansion of macro ‘ASSERTm’
  389 |                 ASSERTm(buf, set_from_string(s.value, s, inputs[i]));
      |                 ^~~~~~~
test/option_parser.c:386:14: note: ‘buf’ declared here
  386 |         char buf[50];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
test/option_parser.c: In function ‘test_string_to_list_invalid’:
test/greatest.h:589:27: warning: storing the address of local variable ‘buf’ in ‘greatest_info.msg’ [-Wdangling-pointer=]
  589 |         greatest_info.msg = MSG;                                        \
test/greatest.h:472:23: note: in expansion of macro ‘GREATEST_FAILm’
  472 |         if ((COND)) { GREATEST_FAILm(MSG); }                            \
      |                       ^~~~~~~~~~~~~~
test/greatest.h:1189:24: note: in expansion of macro ‘GREATEST_ASSERT_FALSEm’
 1189 | #define ASSERT_FALSEm  GREATEST_ASSERT_FALSEm
      |                        ^~~~~~~~~~~~~~~~~~~~~~
test/option_parser.c:432:17: note: in expansion of macro ‘ASSERT_FALSEm’
  432 |                 ASSERT_FALSEm(buf, set_from_string(&val, s, invalid_inputs[i]));
      |                 ^~~~~~~~~~~~~
test/option_parser.c:428:14: note: ‘buf’ declared here
  428 |         char buf[256];
      |              ^~~
test/greatest.h:296:26: note: ‘greatest_info’ declared here
  296 | extern greatest_run_info greatest_info;
      |                          ^~~~~~~~~~~~~
At top level:
cc1: note: unrecognized command-line option ‘-Wno-gnu-zero-variadic-macro-arguments’ may have been intended to silence earlier diagnostics
cc -o test/queues.o -c test/queues.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/rules.o -c test/rules.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/setting.o -c test/setting.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/settings_data.o -c test/settings_data.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/test.o -c test/test.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/utils.o -c test/utils.c -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP
cc -o test/test test/dbus.o test/draw.o test/dunst.o test/helpers.o test/icon-lookup.o test/icon.o test/ini.o test/input.o test/log.o test/markup.o test/menu.o test/misc.o test/notification.o test/option_parser.o test/queues.o test/rules.o test/setting.o test/settings_data.o test/test.o test/utils.o src/output.o src/settings.o src/wayland/foreign_toplevel.o src/wayland/libgwater-wayland.o src/wayland/pool-buffer.o src/wayland/wl.o src/wayland/wl_output.o src/x11/screen.o src/x11/x.o -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"v1.9.2-26-g7608b97\" -DSYSCONFDIR=\"/usr/local/etc/xdg\"  -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os -DENABLE_WAYLAND   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/cairo -I/usr/include/pixman-1 -pthread  -MMD -MP -lm -lrt  -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -lcairo -lX11 -lXinerama -lXext -lXrandr -lXss -lnotify -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lwayland-client -lwayland-cursor 
valgrind \
        --suppressions=.valgrind.suppressions \
        --leak-check=full \
        --show-leak-kinds=definite \
        --errors-for-leak-kinds=definite \
        --num-callers=40 \
        --error-exitcode=123 \
        ./test/test -v
==109216== Memcheck, a memory error detector
==109216== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==109216== Using Valgrind-3.21.0 and LibVEX; rerun with -h for copyright info
==109216== Command: ./test/test -v
==109216== 

* Suite suite_utils:
PASS test_string_replace_char:  (5243 ticks, 0.005 sec)
PASS test_string_replace_all:  (7880 ticks, 0.008 sec)
PASS test_string_append:  (7653 ticks, 0.008 sec)
PASS test_string_strip_quotes:  (2852 ticks, 0.003 sec)
PASS test_string_strip_delimited:  (5622 ticks, 0.006 sec)
PASS test_string_to_path:  (5785 ticks, 0.006 sec)
PASS test_string_to_time:  (4850 ticks, 0.005 sec)

7 tests - 7 passed, 0 failed, 0 skipped (76757 ticks, 0.077 sec)

* Suite suite_option_parser:
PASS test_cmdline_get_string:  (8348 ticks, 0.008 sec)
PASS test_cmdline_get_list:  (8728 ticks, 0.009 sec)
PASS test_cmdline_get_path:  (3056 ticks, 0.003 sec)
PASS test_cmdline_get_int:  (4473 ticks, 0.004 sec)
PASS test_cmdline_get_double:  (19088 ticks, 0.019 sec)
PASS test_cmdline_get_bool:  (2850 ticks, 0.003 sec)
PASS test_cmdline_create_usage:  (3638 ticks, 0.004 sec)
PASS test_string_to_int:  (4436 ticks, 0.004 sec)
PASS test_string_to_int_invalid:  (3584 ticks, 0.004 sec)
PASS test_string_to_double:  (3593 ticks, 0.004 sec)
PASS test_string_to_double_invalid:  (3723 ticks, 0.004 sec)
PASS test_string_to_enum:  (2195 ticks, 0.002 sec)
PASS test_string_to_enum_invalid:  (2901 ticks, 0.003 sec)
PASS test_string_to_boolean:  (3064 ticks, 0.003 sec)
PASS test_string_to_boolean_invalid:  (2815 ticks, 0.003 sec)
PASS test_string_to_list:  (5433 ticks, 0.005 sec)
PASS test_string_to_list_invalid:  (4922 ticks, 0.005 sec)
PASS test_string_to_time:  (3662 ticks, 0.004 sec)
PASS test_string_to_time_invalid:  (3670 ticks, 0.004 sec)
PASS test_string_to_path:  (5285 ticks, 0.005 sec)
PASS test_string_to_sepcolor:  (4913 ticks, 0.005 sec)
PASS test_string_to_sepcolor_invalid:  (3883 ticks, 0.004 sec)
PASS test_enum_size:  (446 ticks, 0.000 sec)
PASS test_string_to_length:  (6651 ticks, 0.007 sec)
PASS test_string_to_length_invalid:  (5009 ticks, 0.005 sec)

25 tests - 25 passed, 0 failed, 0 skipped (132502 ticks, 0.133 sec)

* Suite suite_notification:
PASS test_notification_is_duplicate:  (13447 ticks, 0.013 sec)
PASS test_notification_replace_single_field:  (12066 ticks, 0.012 sec)
PASS test_notification_referencing:  (2089 ticks, 0.002 sec)
PASS test_notification_icon_scaling_toosmall:  (3809502 ticks, 3.810 sec)
PASS test_notification_icon_scaling_toolarge:  (30682 ticks, 0.031 sec)
PASS test_notification_icon_scaling_notconfigured:  (26998 ticks, 0.027 sec)
PASS test_notification_icon_scaling_notneeded:  (26443 ticks, 0.026 sec)
PASS test_notification_format_message:  (2763 ticks, 0.003 sec)
PASS test_notification_format_message:  (588 ticks, 0.001 sec)
PASS test_notification_format_message:  (182 ticks, 0.000 sec)
PASS test_notification_format_message:  (1224 ticks, 0.001 sec)
PASS test_notification_format_message:  (435 ticks, 0.000 sec)
PASS test_notification_format_message:  (1086 ticks, 0.001 sec)
PASS test_notification_format_message:  (307 ticks, 0.000 sec)
PASS test_notification_format_message:  (266 ticks, 0.000 sec)
PASS test_notification_format_message:  (270 ticks, 0.000 sec)
PASS test_notification_format_message:  (1272 ticks, 0.001 sec)
PASS test_notification_maxlength:  (6049 ticks, 0.006 sec)

18 tests - 18 passed, 0 failed, 0 skipped (3943050 ticks, 3.943 sec)

* Suite suite_markup:
PASS test_markup_strip:  (3065 ticks, 0.003 sec)
PASS helper_markup_strip_a:  (4266 ticks, 0.004 sec)
PASS helper_markup_strip_a:  (568 ticks, 0.001 sec)
PASS helper_markup_strip_a:  (735 ticks, 0.001 sec)
PASS helper_markup_strip_a:  (777 ticks, 0.001 sec)
PASS helper_markup_strip_a:  (1844 ticks, 0.002 sec)
PASS helper_markup_strip_a:  (599 ticks, 0.001 sec)
PASS helper_markup_strip_a:  (878 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (3191 ticks, 0.003 sec)
PASS helper_markup_strip_img:  (1674 ticks, 0.002 sec)
PASS helper_markup_strip_img:  (2308 ticks, 0.002 sec)
PASS helper_markup_strip_img:  (1128 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (828 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (569 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (724 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (538 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (580 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (613 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (666 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (615 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (1115 ticks, 0.001 sec)
PASS helper_markup_strip_img:  (1417 ticks, 0.001 sec)
PASS test_markup_transform:  (12653 ticks, 0.013 sec)

23 tests - 23 passed, 0 failed, 0 skipped (44843 ticks, 0.045 sec)

* Suite suite_misc:
PASS assert_version_number:  (441 ticks, 0.000 sec)

1 test - 1 passed, 0 failed, 0 skipped (758 ticks, 0.001 sec)

* Suite suite_icon:
Icon path: /home/friso/git-repos/dunst/test/data
PASS test_get_path_from_icon_null:  (1286 ticks, 0.001 sec)
PASS test_get_path_from_icon_name_full:  (2208 ticks, 0.002 sec)
PASS test_icon_size_clamp_not_necessary:  (1144 ticks, 0.001 sec)
PASS test_icon_size_clamp_too_small:  (923 ticks, 0.001 sec)
PASS test_icon_size_clamp_not_necessary:  (449 ticks, 0.000 sec)
PASS test_icon_size_clamp_too_big:  (933 ticks, 0.001 sec)
PASS test_icon_size_clamp_too_small_then_too_big:  (1021 ticks, 0.001 sec)
PASS test_icon_size_clamp_too_small:  (462 ticks, 0.000 sec)
PASS test_icon_size_clamp_not_necessary:  (466 ticks, 0.000 sec)
PASS test_icon_size_clamp_not_necessary:  (461 ticks, 0.000 sec)
PASS test_icon_size_clamp_too_big:  (483 ticks, 0.000 sec)

11 tests - 11 passed, 0 failed, 0 skipped (20519 ticks, 0.021 sec)

* Suite suite_queues:
PASS test_datachange_beginning_empty:  (3343 ticks, 0.003 sec)
PASS test_datachange_endless:  (150469 ticks, 0.150 sec)
PASS test_datachange_endless_agethreshold:  (3546 ticks, 0.004 sec)
PASS test_datachange_agethreshold_at_second:  (6334 ticks, 0.006 sec)
PASS test_datachange_queues:  (5479 ticks, 0.005 sec)
PASS test_datachange_ttl:  (5389 ticks, 0.005 sec)
PASS test_queue_history_clear:  (4621 ticks, 0.005 sec)
PASS test_queue_history_overfull:  (8223 ticks, 0.008 sec)
PASS test_queue_history_pushall:  (12280 ticks, 0.012 sec)
PASS test_queue_history_remove_by_id:  (6091 ticks, 0.006 sec)
PASS test_queue_init:  (1143 ticks, 0.001 sec)
PASS test_queue_insert_id_invalid:  (3001 ticks, 0.003 sec)
PASS test_queue_insert_id_replacement:  (8606 ticks, 0.009 sec)
PASS test_queue_insert_id_valid_newid:  (2328 ticks, 0.002 sec)
PASS test_queue_length:  (4712 ticks, 0.005 sec)
PASS test_queue_notification_close:  (7447 ticks, 0.007 sec)
PASS test_queue_notification_close_histignore:  (7009 ticks, 0.007 sec)
PASS test_queue_notification_skip_display:  (3493 ticks, 0.003 sec)
PASS test_queue_notification_skip_display_redisplayed:  (5670 ticks, 0.006 sec)
PASS test_queue_notification_skip_display_redisplayed_by_random_id:  (11551 ticks, 0.012 sec)
PASS test_queue_stacking:  (8339 ticks, 0.008 sec)
PASS test_queue_stacktag:  (7840 ticks, 0.008 sec)
queue 0
PASS test_queue_different_stacktag:  (5833 ticks, 0.006 sec)
PASS test_queue_stacktag_different_appid:  (5767 ticks, 0.006 sec)
PASS test_queue_teardown:  (2484 ticks, 0.002 sec)
PASS test_queue_timeout:  (7182 ticks, 0.007 sec)
PASS test_queues_update_fullscreen:  (6658 ticks, 0.007 sec)
PASS test_queues_update_paused:  (6629 ticks, 0.007 sec)
PASS test_queues_update_seep_showlowurg:  (5595 ticks, 0.006 sec)
PASS test_queues_update_seeping:  (16384 ticks, 0.016 sec)
PASS test_queues_update_xmore:  (7789 ticks, 0.008 sec)
PASS test_queues_timeout_before_paused:  (2936 ticks, 0.003 sec)
PASS test_queue_find_by_id:  (4329 ticks, 0.004 sec)
PASS test_queue_no_sort_and_pause:  (10142 ticks, 0.010 sec)
PASS test_queue_get_history:  (5110 ticks, 0.005 sec)

35 tests - 35 passed, 0 failed, 0 skipped (372538 ticks, 0.373 sec)

* Suite suite_dunst:
PASS test_dunst_status:  (462 ticks, 0.000 sec)

1 test - 1 passed, 0 failed, 0 skipped (792 ticks, 0.001 sec)

* Suite suite_log:
PASS test_log_level:  (4437 ticks, 0.004 sec)
PASS test_log_level:  (1105 ticks, 0.001 sec)
PASS test_log_level:  (700 ticks, 0.001 sec)
PASS test_log_level:  (718 ticks, 0.001 sec)
PASS test_log_level:  (811 ticks, 0.001 sec)
PASS test_log_level:  (840 ticks, 0.001 sec)

6 tests - 6 passed, 0 failed, 0 skipped (10063 ticks, 0.010 sec)

* Suite suite_menu:
PASS test_extract_urls_from_empty_string:  (1730 ticks, 0.002 sec)
PASS test_extract_urls_from_no_urls_string:  (940 ticks, 0.001 sec)
PASS test_extract_urls_from_one_url_string:  (7219 ticks, 0.007 sec)
PASS test_extract_urls_from_two_url_string:  (1147 ticks, 0.001 sec)
PASS test_extract_urls_from_one_url_port:  (977 ticks, 0.001 sec)
PASS test_extract_urls_from_one_url_path:  (1007 ticks, 0.001 sec)
PASS test_extract_urls_from_one_url_anchor:  (997 ticks, 0.001 sec)

7 tests - 7 passed, 0 failed, 0 skipped (15812 ticks, 0.016 sec)

* Suite suite_settings_data:
PASS test_names_valid:  (4148 ticks, 0.004 sec)
PASS test_description_valid:  (3726 ticks, 0.004 sec)
PASS test_type_valid:  (2541 ticks, 0.003 sec)
PASS test_section_valid:  (3710 ticks, 0.004 sec)
PASS test_default_value_valid:  (3746 ticks, 0.004 sec)
PASS test_value_non_null:  (2606 ticks, 0.003 sec)
PASS test_valid_parser_and_data_per_type:  (6213 ticks, 0.006 sec)

7 tests - 7 passed, 0 failed, 0 skipped (28719 ticks, 0.029 sec)

* Suite suite_dbus:
PASS test_dbus_init:  (1082955 ticks, 1.083 sec)
PASS test_get_fdn_daemon_info:  (275643 ticks, 0.276 sec)
PASS test_dbus_cb_dunst_Properties_Get:  (2829 ticks, 0.003 sec)
PASS test_dbus_cb_dunst_Properties_Set:  (66826 ticks, 0.067 sec)
PASS test_empty_notification:  (37729 ticks, 0.038 sec)
PASS test_basic_notification:  (25253 ticks, 0.025 sec)
PASS test_invalid_notification:  (1721 ticks, 0.002 sec)
PASS test_hint_transient:  (144262 ticks, 0.144 sec)
PASS test_hint_progress:  (117788 ticks, 0.118 sec)
PASS test_hint_icons:  (17686 ticks, 0.018 sec)
PASS test_hint_category:  (19469 ticks, 0.019 sec)
PASS test_hint_desktop_entry:  (21867 ticks, 0.022 sec)
PASS test_hint_urgency:  (86139 ticks, 0.086 sec)
PASS test_hint_raw_image:  (182611 ticks, 0.183 sec)
PASS test_dbus_notify_colors:  (19835 ticks, 0.020 sec)
PASS test_server_caps:  (19714 ticks, 0.020 sec)
PASS test_server_caps:  (12986 ticks, 0.013 sec)
PASS test_server_caps:  (14749 ticks, 0.015 sec)
PASS test_close_and_signal:  (46378 ticks, 0.046 sec)
PASS test_signal_actioninvoked:  (33859 ticks, 0.034 sec)
PASS test_signal_length_propertieschanged:  (44208 ticks, 0.044 sec)
PASS test_timeout_overflow:  (21580 ticks, 0.022 sec)
PASS test_override_dbus_timeout:  (19348 ticks, 0.019 sec)
PASS test_match_dbus_timeout:  (22807 ticks, 0.023 sec)
PASS test_timeout:  (22683 ticks, 0.023 sec)
PASS assert_methodlists_sorted:  (1300 ticks, 0.001 sec)
PASS test_dbus_teardown:  (49598 ticks, 0.050 sec)

27 tests - 27 passed, 0 failed, 0 skipped (2663689 ticks, 2.664 sec)

* Suite suite_setting:
PASS test_dunstrc_markup:  (11261 ticks, 0.011 sec)
PASS test_dunstrc_nomarkup:  (10215 ticks, 0.010 sec)
PASS test_dunstrc_defaults:  (25899 ticks, 0.026 sec)

3 tests - 3 passed, 0 failed, 0 skipped (48192 ticks, 0.048 sec)

* Suite suite_icon_lookup:
PASS test_load_theme_from_dir:  (267335 ticks, 0.267 sec)
PASS test_find_icon:  (258218 ticks, 0.258 sec)
PASS test_find_path:  (254120 ticks, 0.254 sec)
PASS test_new_icon_overrides_raw_icon:  (359501 ticks, 0.360 sec)

4 tests - 4 passed, 0 failed, 0 skipped (1140358 ticks, 1.140 sec)

* Suite suite_draw:
PASS test_layout_render_gaps:  (2890126 ticks, 2.890 sec)
PASS test_layout_from_notification_icon_off:  (2733 ticks, 0.003 sec)
PASS test_layout_from_notification:  (2099 ticks, 0.002 sec)
PASS test_calculate_dimensions_height_no_gaps:  (13264 ticks, 0.013 sec)
PASS test_layout_from_notification_no_icon:  (2208 ticks, 0.002 sec)
PASS test_layout_render_no_gaps:  (15619 ticks, 0.016 sec)
PASS test_calculate_dimensions_height_gaps:  (13169 ticks, 0.013 sec)

7 tests - 7 passed, 0 failed, 0 skipped (2945040 ticks, 2.945 sec)

* Suite suite_rules:
PASS test_pattern_match:  (16080 ticks, 0.016 sec)
PASS test_pattern_match:  (41596 ticks, 0.042 sec)

2 tests - 2 passed, 0 failed, 0 skipped (58451 ticks, 0.058 sec)

* Suite suite_input:
PASS test_get_notification_clickable_height_middle:  (1624 ticks, 0.002 sec)
PASS test_get_notification_clickable_height_first:  (1656 ticks, 0.002 sec)
PASS test_notification_at:  (8092 ticks, 0.008 sec)
PASS test_get_notification_clickable_height_gaps:  (2014 ticks, 0.002 sec)
PASS test_get_notification_clickable_height_last:  (1482 ticks, 0.001 sec)

5 tests - 5 passed, 0 failed, 0 skipped (18522 ticks, 0.019 sec)

Total: 189 tests (11556467 ticks, 11.556 sec), 2392 assertions
Pass: 189, fail: 0, skip: 0.
==109216== 
==109216== HEAP SUMMARY:
==109216==     in use at exit: 985,879 bytes in 13,464 blocks
==109216==   total heap usage: 140,160 allocs, 126,696 frees, 30,948,597 bytes allocated
==109216== 
==109216== LEAK SUMMARY:
==109216==    definitely lost: 0 bytes in 0 blocks
==109216==    indirectly lost: 4,679 bytes in 210 blocks
==109216==      possibly lost: 4,822 bytes in 15 blocks
==109216==    still reachable: 648,236 bytes in 3,005 blocks
==109216==         suppressed: 310,110 bytes in 10,070 blocks
==109216== Reachable blocks (those to which a pointer was found) are not shown.
==109216== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==109216== 
==109216== For lists of detected and suppressed errors, rerun with: -s
==109216== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 5 from 5)
```

</details>

--- a/config.mk
+++ b/config.mk
@@ -42,7 +42,7 @@ endif
 
 # flags
 DEFAULT_CPPFLAGS = -Wno-gnu-zero-variadic-macro-arguments -D_DEFAULT_SOURCE -DVERSION=\"${VERSION}\" -DSYSCONFDIR=\"${SYSCONFDIR}\"
-DEFAULT_CFLAGS   = -g -std=gnu99 -pedantic -Wall -Wno-overlength-strings -Os ${ENABLE_WAYLAND} ${EXTRA_CFLAGS}
+DEFAULT_CFLAGS   = -g -std=gnu11 -pedantic -Wall -Wno-overlength-strings -Os ${ENABLE_WAYLAND} ${EXTRA_CFLAGS}
 DEFAULT_LDFLAGS  = -lm -lrt
 
 CPPFLAGS_DEBUG := -DDEBUG_BUILD

--- a/src/icon-lookup.c
+++ b/src/icon-lookup.c
@@ -157,7 +157,7 @@ int load_icon_theme_from_dir(const char *icon_dir, const char *subdir_theme) {
 // a list of directories where icon themes might be located
 GPtrArray *theme_path = NULL;
 
-void get_theme_path() {
+void get_theme_path(void) {
         theme_path = g_ptr_array_new_full(5, g_free);
         const char *home = g_get_home_dir();
         g_ptr_array_add(theme_path, g_build_filename(home, ".icons", NULL));
@@ -208,7 +208,7 @@ void finish_icon_theme(struct icon_theme *theme) {
         g_free(theme->dirs);
 }
 
-void free_all_themes() {
+void free_all_themes(void) {
         g_free(default_themes_index);
         default_themes_index = NULL;
         default_themes_count = 0;

--- a/src/icon-lookup.h
+++ b/src/icon-lookup.h
@@ -76,7 +76,7 @@ char *find_icon_path(const char *name, int size);
 /**
  * Free all icon themes.
  */
-void free_all_themes();
+void free_all_themes(void);
 
 #endif
 /* vim: set ft=c tabstop=8 shiftwidth=8 expandtab textwidth=0: */

--- a/src/option_parser.c
+++ b/src/option_parser.c
@@ -403,7 +403,7 @@ bool set_rule(struct setting setting, char* value, char* section) {
         return set_rule_value(r, setting, value);
 }
 
-void set_defaults() {
+void set_defaults(void) {
         for (int i = 0; i < G_N_ELEMENTS(allowed_settings); i++) {
                 // FIXME Rule settings can only have a default if they have an
                 // working entry in the settings struct as well. Make an

--- a/src/option_parser.h
+++ b/src/option_parser.h
@@ -14,7 +14,7 @@ int string_parse_enum(const void* data, const char *s, void * ret);
 int string_parse_sepcolor(const void *data, const char *s, void *ret);
 int string_parse_bool(const void *data, const char *s, void *ret);
 
-void set_defaults();
+void set_defaults(void);
 void save_settings(struct ini *ini);
 
 void cmdline_load(int argc, char *argv[]);

--- a/src/output.c
+++ b/src/output.c
@@ -57,7 +57,7 @@ const struct output output_wl = {
 };
 #endif
 
-const struct output* get_x11_output() {
+const struct output* get_x11_output(void) {
         const struct output* output = &output_x11;
         if (output->init()) {
                 return output;
@@ -67,7 +67,7 @@ const struct output* get_x11_output() {
 }
 
 #ifdef ENABLE_WAYLAND
-const struct output* get_wl_output() {
+const struct output* get_wl_output(void) {
         const struct output* output = &output_wl;
         if (output->init()) {
                 return output;

--- a/src/rules.c
+++ b/src/rules.c
@@ -178,7 +178,7 @@ static inline bool rule_field_matches_string(const char *value, const char *patt
                         return false;
                 }
 
-                for (int i = 0; ; i++) {
+                while (true) {
                         if (regexec(&regex, value, 0, NULL, 0))
                                 break;
                         regfree(&regex);

--- a/src/settings.c
+++ b/src/settings.c
@@ -60,7 +60,7 @@ static int is_drop_in(const struct dirent *dent) {
  * The result @e must @e not be freed! The array is cached in a static variable,
  * so it is OK to call this again instead of caching its return value.
  */
-static GPtrArray *get_xdg_conf_basedirs() {
+static GPtrArray *get_xdg_conf_basedirs(void) {
         GPtrArray *arr = g_ptr_array_new_full(4, g_free);
         g_ptr_array_add(arr, g_build_filename(g_get_user_config_dir(), "dunst", NULL));
 
@@ -151,7 +151,7 @@ FILE *fopen_conf(char * const path)
         return f;
 }
 
-void settings_init() {
+void settings_init(void) {
         static bool init_done = false;
         if (!init_done) {
                 LOG_D("Initializing settings");

--- a/src/wayland/foreign_toplevel.c
+++ b/src/wayland/foreign_toplevel.c
@@ -12,10 +12,6 @@
 
 struct wl_list toplevel_list;
 
-static void noop() {
-        // This space intentionally left blank
-}
-
 static void copy_state(struct toplevel_state *current,
                 struct toplevel_state *pending) {
         if (!(pending->state & TOPLEVEL_STATE_INVALID)) {
@@ -98,9 +94,19 @@ static void toplevel_handle_closed(void *data,
         zwlr_foreign_toplevel_handle_v1_destroy(zwlr_toplevel);
 }
 
+static void toplevel_handle_title(void *data,
+                struct zwlr_foreign_toplevel_handle_v1 *zwlr_foreign_toplevel_handle_v1,
+                const char *title) {
+}
+
+static void toplevel_handle_app_id(void *data,
+                struct zwlr_foreign_toplevel_handle_v1 *zwlr_foreign_toplevel_handle_v1,
+                const char *title) {
+}
+
 static const struct zwlr_foreign_toplevel_handle_v1_listener toplevel_impl = {
-        .title = noop,
-        .app_id = noop,
+        .title = toplevel_handle_title,
+        .app_id = toplevel_handle_app_id,
         .output_enter = toplevel_handle_output_enter,
         .output_leave = toplevel_handle_output_leave,
         .state = toplevel_handle_state,

--- a/src/wayland/wl.c
+++ b/src/wayland/wl.c
@@ -93,11 +93,7 @@ struct wl_ctx {
 
 struct wl_ctx ctx;
 
-static void noop() {
-        // This space intentionally left blank
-}
-
-void set_dirty();
+void set_dirty(void);
 
 static void output_handle_geometry(void *data, struct wl_output *wl_output,
                 int32_t x, int32_t y, int32_t phy_width, int32_t phy_height,
@@ -122,10 +118,14 @@ static void output_handle_scale(void *data, struct wl_output *wl_output,
         wake_up();
 }
 
+static void output_listener_done_handler(void *data, struct wl_output *output) {
+        // do nothing
+}
+
 static const struct wl_output_listener output_listener = {
         .geometry = output_handle_geometry,
         .mode = output_handle_mode,
-        .done = noop,
+        .done = output_listener_done_handler,
         .scale = output_handle_scale,
 };
 
@@ -154,7 +154,7 @@ static void create_output( struct wl_output *wl_output, uint32_t global_name) {
 
         if (recreate_surface) {
                 // We had no outputs, force our surface to redraw
-                set_dirty(ctx.surface);
+                set_dirty();
         }
 }
 
@@ -220,20 +220,34 @@ static void pointer_handle_button(void *data, struct wl_pointer *wl_pointer,
         input_handle_click(button, button_state, ctx.pointer.x, ctx.pointer.y);
 }
 
+static void pointer_handle_leave(void *data, struct wl_pointer *wl_pointer,
+                uint32_t serial, struct wl_surface *surface) {
+}
+
+static void pointer_handle_axis(void *data, struct wl_pointer *wl_pointer,
+                uint32_t time, uint32_t axis, wl_fixed_t value) {
+}
+
 static const struct wl_pointer_listener pointer_listener = {
         .enter = pointer_handle_enter,
-        .leave = noop,
+        .leave = pointer_handle_leave,
         .motion = pointer_handle_motion,
         .button = pointer_handle_button,
-        .axis = noop,
+        .axis = pointer_handle_axis,
 };
+
+static void touch_handle_frame(void *data, struct wl_touch *wl_touch) {
+}
+
+static void touch_handle_cancel(void *data, struct wl_touch *wl_touch) {
+}
 
 static const struct wl_touch_listener touch_listener = {
         .down = touch_handle_down,
         .up = touch_handle_up,
         .motion = touch_handle_motion,
-        .frame = noop,
-        .cancel = noop,
+        .frame = touch_handle_frame,
+        .cancel = touch_handle_cancel,
 };
 
 static void seat_handle_capabilities(void *data, struct wl_seat *wl_seat,
@@ -259,9 +273,12 @@ static void seat_handle_capabilities(void *data, struct wl_seat *wl_seat,
         }
 }
 
+static void seat_handle_name(void *data, struct wl_seat *wl_seat,
+                const char *name) {}
+
 static const struct wl_seat_listener seat_listener = {
         .capabilities = seat_handle_capabilities,
-        .name = noop,
+        .name = seat_handle_name,
 };
 
 
@@ -284,8 +301,8 @@ static const struct wl_surface_listener surface_listener = {
 };
 
 
-static void schedule_frame_and_commit();
-static void send_frame();
+static void schedule_frame_and_commit(void);
+static void send_frame(void);
 
 static void layer_surface_handle_configure(void *data,
                 struct zwlr_layer_surface_v1 *surface,
@@ -367,7 +384,7 @@ static void add_seat_to_idle_handler(struct wl_seat *seat) {
 }
 
 // Warning, can return NULL
-static struct dunst_output *get_configured_output() {
+static struct dunst_output *get_configured_output(void) {
         int n = 0;
         int target_monitor = settings.monitor;
 
@@ -446,7 +463,7 @@ static const struct wl_registry_listener registry_listener = {
         .global_remove = handle_global_remove,
 };
 
-bool wl_init() {
+bool wl_init(void) {
         wl_list_init(&ctx.outputs);
         wl_list_init(&toplevel_list);
         //wl_list_init(&ctx.seats); // TODO multi-seat support
@@ -545,7 +562,7 @@ bool wl_init() {
         return true;
 }
 
-void wl_deinit() {
+void wl_deinit(void) {
         // We need to check if any of these are NULL, since the initialization
         // could have been aborted half way through, or the compositor doesn't
         // support some of these features.
@@ -599,10 +616,10 @@ void wl_deinit() {
         g_water_wayland_source_free(ctx.esrc);
 }
 
-static void schedule_frame_and_commit();
+static void schedule_frame_and_commit(void);
 
 // Draw and commit a new frame.
-static void send_frame() {
+static void send_frame(void) {
         int scale = wl_get_scale();
 
         if (wl_list_empty(&ctx.outputs)) {
@@ -749,7 +766,7 @@ static const struct wl_callback_listener frame_listener = {
         .done = frame_handle_done,
 };
 
-static void schedule_frame_and_commit() {
+static void schedule_frame_and_commit(void) {
         if (ctx.frame_callback) {
                 return;
         }
@@ -763,7 +780,7 @@ static void schedule_frame_and_commit() {
         wl_surface_commit(ctx.surface);
 }
 
-void set_dirty() {
+void set_dirty(void) {
         if (ctx.dirty) {
                 return;
         }

--- a/test/draw.c
+++ b/test/draw.c
@@ -5,7 +5,7 @@
 
 cairo_t *c;
 
-double get_dummy_scale() { return 1; }
+double get_dummy_scale(void) { return 1; }
 
 const struct screen_info* noop_screen(void) {
         static struct screen_info i;

--- a/test/icon-lookup.c
+++ b/test/icon-lookup.c
@@ -7,7 +7,7 @@
 extern const char *base;
 #define ICONPREFIX "data", "icons"
 
-int setup_test_theme(){
+int setup_test_theme(void) {
         char *theme_path = g_build_filename(base, ICONPREFIX,  NULL);
         int theme_index = load_icon_theme_from_dir(theme_path, "theme");
         add_default_theme(theme_index);

--- a/test/option_parser.c
+++ b/test/option_parser.c
@@ -139,7 +139,7 @@ TEST test_string_to_int(void)
         struct setting s;
         s.type = TYPE_INT;
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERTm(buf, set_from_string(&val, s, inputs[i]));
@@ -166,7 +166,7 @@ TEST test_string_to_int_invalid(void)
         s.type = TYPE_INT;
         s.name = "test_int";
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));
@@ -199,7 +199,7 @@ TEST test_string_to_double(void)
         struct setting s;
         s.type = TYPE_DOUBLE;
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERTm(buf, set_from_string(&val, s, inputs[i]));
@@ -225,7 +225,7 @@ TEST test_string_to_double_invalid(void)
         s.type = TYPE_DOUBLE;
         s.name = "test_double";
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));
@@ -263,7 +263,7 @@ TEST test_string_to_boolean(void)
 
         ARRAY_SAME_LENGTH(inputs, results);
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERTm(buf, set_from_string(&val, s, inputs[i]));
@@ -291,7 +291,7 @@ TEST test_string_to_boolean_invalid(void)
                 "else",
         };
 
-        char buf[50];
+        static char buf[50];
 
         for (int i = 0; i < G_N_ELEMENTS(invalid_inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
@@ -312,7 +312,7 @@ TEST test_string_to_enum(void)
         s.parser = string_parse_enum;
         s.parser_data = ellipsize_enum_data;
 
-        char buf[50];
+        static char buf[50];
 
         // do not go until last element, since it's ENUM_END (all 0)
         for (int i = 0; i < G_N_ELEMENTS(ellipsize_enum_data)-1; i++) {
@@ -342,7 +342,7 @@ TEST test_string_to_enum_invalid(void)
                 "else"
         };
 
-        char buf[50];
+        static char buf[50];
 
         for (int i = 0; i < G_N_ELEMENTS(invalid_inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
@@ -383,7 +383,7 @@ TEST test_string_to_list(void)
                 {MOUSE_CLOSE_ALL, MOUSE_CLOSE_CURRENT, MOUSE_CLOSE_ALL, MOUSE_ACTION_END},
         };
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERTm(buf, set_from_string(s.value, s, inputs[i]));
@@ -425,7 +425,7 @@ TEST test_string_to_list_invalid(void)
                 "close_all,invalid,close_current,close_all",
         };
 
-        char buf[256];
+        static char buf[256];
 
         for (int i = 0; i < G_N_ELEMENTS(invalid_inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
@@ -465,7 +465,7 @@ TEST test_string_to_time(void)
 
         ARRAY_SAME_LENGTH(inputs, results);
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERTm(buf, set_from_string(&val, s, inputs[i]));
@@ -495,7 +495,7 @@ TEST test_string_to_time_invalid(void)
                 "s",
         };
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(invalid_inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERT_FALSEm(buf, set_from_string(&val, s, invalid_inputs[i]));
@@ -554,7 +554,7 @@ TEST test_string_to_path(void)
         ARRAY_SAME_LENGTH(inputs, results);
         ARRAY_SAME_LENGTH(inputs, results2);
 
-        char buf[256];
+        static char buf[256];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i", i);
                 ASSERTm(buf, set_from_string(&val, s, inputs[i]));
@@ -602,7 +602,7 @@ TEST test_string_to_sepcolor(void)
 
         ARRAY_SAME_LENGTH(inputs, results);
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i. Expected %i, got %i", i, results[i].type, val.type);
                 ASSERTm(buf, set_from_string(&val, s, inputs[i]));
@@ -638,7 +638,7 @@ TEST test_string_to_sepcolor_invalid(void)
                 /* "#AB123C123212", */
         };
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i.", i);
                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));
@@ -682,7 +682,7 @@ TEST test_string_to_length(void)
 
         ARRAY_SAME_LENGTH(inputs, results);
 
-        char buf[500];
+        static char buf[500];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i.", i);
                 ASSERTm(buf, set_from_string(&val, s, inputs[i]));
@@ -726,7 +726,7 @@ TEST test_string_to_length_invalid(void)
                 "(123, 122)", // invalid order
         };
 
-        char buf[50];
+        static char buf[50];
         for (int i = 0; i < G_N_ELEMENTS(inputs); i++) {
                 sprintf(buf, "Failed in round %i.", i);
                 ASSERT_FALSEm(buf, set_from_string(&val, s, inputs[i]));

--- a/test/queues.c
+++ b/test/queues.c
@@ -1061,7 +1061,7 @@ TEST test_queue_get_history(void)
 }
 
 
-void print_queues() {
+void print_queues(void) {
         printf("\nQueues:\n");
         for (GList *iter = g_queue_peek_head_link(QUEUE_WAIT); iter;
                         iter = iter->next) {

